packages/workflow/src/Cron.ts

```diff
@@ -0,0 +1,68 @@
+interface BaseTriggerTime<T extends string> {
+	mode: T;
+}
+
+type CronExpression = string;
+interface CustomTrigger extends BaseTriggerTime<'custom'> {
+	cronExpression: CronExpression;
+}
+
+interface EveryXTrigger<U extends string> extends BaseTriggerTime<'everyX'> {
```

Comment:
Let's tighten the first constraint to a union of the admissible values?

Reply:
same as above. the exported interface takes care of the admissible values.

Reply:
https://github.com/n8n-io/n8n/pull/3887#discussion_r949221629